server/src/api/listings.ts

```diff
@@ -25,19 +25,42 @@ import {listingService} from '../services';
 
 const listingRouter = Router();
 
+/**
+ * Handles the get request for retrieving all listings that satisfy the query
+ * parameters.
+ * Querying by ids is a restricted query, once the id field is specified, requester
+ * would not be allowed to filter by any other query params.
+ * Otherwise, it would be a Bad Request.
+ */
 listingRouter.get(
   '/',
   async (req: Request, res: Response, next: NextFunction) => {
     try {
+      const {ids: idsStr} = req.query;
+      if (idsStr !== undefined) {
+        if (Object.keys(req.query).length > 1) {
+          res.sendStatus(400);
+          return;
+        }
+
+        const ids = (idsStr as string).split(',').map(idStr => {
+          const id = Number(idStr);
+          if (Number.isNaN(id)) {
+            throw new Error(`Invalid listingId ${idStr}`);
+          }
+          return id;
+        });
+        const listings = await listingService.getAllListingsWithIds(ids);
+        res.json(listings);
+        return;
+      }
+
       const filters = Object.keys(req.query).map(key => ({
         property: key,
-        value:
-          key === 'customerId' || key === 'merchantId'
```

Comment:
you removed customerId. is it expected?

Reply:
@ifhuang Yep it's not needed anymore because initially I wanted to query Listings by `customerid`, but since `customerId` is not a field it makes the handling of this query quite messy. So now api consumers should first query for Commits with `customerId` and then query for Listings using the `listingId`s from the Commits.